app/src/org/commcare/android/tasks/DataPullTask.java

```diff
@@ -124,6 +121,8 @@ public DataPullTask(String username, String password, String server, String keyP
         this.password = password;
         this.c = c;
         this.taskId = DATA_PULL_TASK_ID;
+
+        TAG = DataPullTask.class.getSimpleName();

```

Comment:
The error logging occurs in the parent class, but we want the TAG to be set to the child class's name so in most cases the extending class sets the TAG in its constructor. There are some cases where the extending class doesn't provide its own constructor (see https://github.com/dimagi/commcare-odk/blob/master/app/src/org/commcare/android/tasks/MultimediaInflaterTask.java#L17), so I was lazy and let the parent constructor initialize TAG to the general "CommCareTask". This pattern doesn't allow for the use of 'final'.

If you would like I can add explicit constructors to all the abstract extensions of CommCareTask so that the TAG field is set correctly.